metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -352,6 +386,43 @@ object MetalsEnrichments
         }
         AbsolutePath(out)
       }
+
+      // prevent inifinity loop
+      if (retryCount > 5) {
+        throw new Exception(s"Unable to save $path in workspace")
+      } else if (path.toNIO.getFileSystem == workspace.toNIO.getFileSystem) {
+        path
+      } else {
+        path.jarPath match {
+          case Some(jar) =>
+            val jarDir =
+              workspace.resolve(Directories.dependencies).resolve(jar.filename)
+            val out = jarDir.resolveZipPath(path.toNIO)
+            val jarMetaFile = jarDir.resolve("jar.meta")
```

Comment:
Maybe let's add a dot so that it's a hidden file in some systems?
```suggestion
            val jarMetaFile = jarDir.resolve(".jar.meta")
```